cmd/pitr/pxc/pxc.go

```diff
@@ -187,12 +208,80 @@ func GetPXCLastHost(pxcServiceName string) (string, error) {
 		}
 	}
 	if len(lastHost) == 0 {
-		return "", errors.New("cant find host")
+		return "", errors.New("can't find host")
 	}
 
 	return lastHost, nil
 }
 
+func GetPXCOldestBinlogHost(pxcServiceName, user, pass string) (string, error) {
+	cmd := exec.Command("peer-list", "-on-start=/usr/bin/get-pxc-state", "-service="+pxcServiceName)
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return "", errors.Wrap(err, "get peer-list output")
+	}
+	nodes := strings.Split(string(out), "node:")
+
+	var oldestHost string
+	var oldestTS int64
+	for _, node := range nodes {
+		if strings.Contains(node, "wsrep_ready:ON:wsrep_connected:ON:wsrep_local_state_comment:Synced:wsrep_cluster_status:Primary") {
+			nodeArr := strings.Split(node, ":")
+			db, err := NewPXC(nodeArr[0], user, pass)
+			if err != nil {
+				log.Printf("ERROR: creating connection for host %s: %v", nodeArr[0], err)
+				continue
+			}
+			list, err := db.GetBinLogNamesList()
+			if err != nil {
+				log.Printf("ERROR: get binlog list for host %s: %v", nodeArr[0], err)
+				db.Close()
+				continue
+			}
+			if len(list) == 0 {
+				log.Printf("ERROR: get binlog list for host %s: no binlogs found", nodeArr[0])
+				db.Close()
+				continue
+			}
+			var binlogTime int64
+			for _, binlogName := range list {
+				ts, err := db.GetBinLogFirstTimestamp(binlogName)
+				if err != nil {
+					log.Printf("ERROR: get binlog first timestamp for binlog %s host %s: %v", binlogName, nodeArr[0], err)
+					continue
+				}
+				binlogTime, err = strconv.ParseInt(ts, 10, 64)
+				if err != nil {
+					log.Printf("ERROR: parse timestamp for binlog %s host %s: %v", binlogName, nodeArr[0], err)
+					continue
+				}
+			}
+			if binlogTime == 0 {
+				log.Printf("ERROR: get binlog oldest timestamp for host %s: no binlogs timestamp found", nodeArr[0])
+				db.Close()
+				continue
+			}
+
+			if oldestTS > 0 && binlogTime < oldestTS {
+				oldestHost = nodeArr[0]
+				oldestTS = binlogTime
+			}
+
+			if len(oldestHost) == 0 {
```

Comment:
we can move this logic to the `if` statement above

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/761/commits/f88ca5e04a895030dc7ef74f1397cecb900c20a1